### PR TITLE
Replace all weird chars when creating chart ids

### PR DESCRIPTION
**Pull request by gomoripeti**
In quoted atoms any characters can occure and Elixir makes use of many
of them.

### CODE REVIEW

File: priv/app/graph.jsx [hunk 1]
```diff
@@ -152,17 +152,12 @@ export default class Graph extends React.Component {
...
-    // And "*" is not a valid character too for an ID.
-    if (arity === "*") {
-      arity = "x";
-    }
-
-    return `chart_${safe_module}_${this.props.mfa[1]}_${arity}`;
+    // Characters that have special meaning in CSS selectors are not safe in an ID.
+    // (eg ':', '.', '?', '*' or single quote itself)
+    // (A very problematic example: "'Elixir.List':'keymember?'/*")
+    return "chart_" + formatted_mfa.replace(/[^A-Za-z0-9_-]/g, "-");
```
Comment (afronski) [inline priv/app/graph.jsx]: *Nitpicking*: `-` in pattern is superfluous (we will replace it anyway via `-`) and if required - should be escaped
.

File: priv/app/graph.jsx [hunk 2]
```diff
@@ -152,17 +152,12 @@ export default class Graph extends React.Component {
   }
 
   chartId() {
```
Comment (afronski) [inline priv/app/graph.jsx]: *Nitpicking*: maybe we can add some client-side tests for it?

File: priv/app/graph.jsx (hunk 1 above)
Comment (gomoripeti) [inline priv/app/graph.jsx]: true. maybe my intent was to explicitly list characters that are ok. I was also thinking about escaping it but then realised that this is not enough to have unique ids for different mfas ("m:'f?'/0" still maps to that same id as "m:'f!'/0" - obviously)

File: priv/app/graph.jsx (hunk 2 above)
Comment (gomoripeti) [inline priv/app/graph.jsx]: good suggestion, I'll add tests

**Pull request merged**